buildSrc/src/main/java/com/wire/tasks/CarthageTask.kt

```diff
@@ -0,0 +1,50 @@
+package com.wire.tasks
+import org.gradle.api.DefaultTask
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.Optional
+import org.gradle.api.tasks.TaskAction
+import org.gradle.api.tasks.options.Option
+import com.wire.plugin.CarthageParameters
+
+abstract class CarthageTask : DefaultTask() {
+
+    init {
+        description = "Run Carthage"
+        group = "wire"
+    }
+
+    @get:Input
+    @get:Option(option = "carthageParameters", description = "The set of parameters to run Carthage with")
+    abstract val parameters: Property<CarthageParameters>
+
+    @get:Input
+    @get:Option(option = "tag", description = "A Tag to be used for debug")
+    @get:Optional
+    abstract val tag: Property<String>
+
+    @TaskAction
+    fun carthageAction() {
+        val prettyTag = tag.orNull?.let { "[$it]" } ?: ""
+
+        val platforms = parameters.get().platforms.joinToString(" ")
+        logger.lifecycle("$prettyTag carthage invocation: carthage ${parameters.get().command.commandString} --cache-builds --platform $platforms ${if(parameters.get().useXCFrameworks) "--use-xcframeworks" else ""}")
+
+        project.exec {
```

Comment:
I guess it will just blow up if `carthage` is not installed (_e.g._ Windows, or Android devs that are coming aboard the project). But being the only one that doesn't use a Mac all the time, I can fix this later myself. I'll try to find the best place to make the cut.

Reply:
These tasks don't run automatically yet, but certainly a good point.